NMCA_001/NMCA_001_script.lua

```diff
@@ -0,0 +1,767 @@
+    Ship:SetReclaimable(false)
+
+    -- Mark objectives as failed
+    for k, v in AssignedObjectives do
+        if(v and v.Active) then
+            v:ManualResult(false)
+        end
+    end
+
+	ScenarioFramework.CDRDeathNISCamera(Ship)
+
+	local UEFUnits = GetUnitsInRect(ScenarioUtils.AreaToRect('M1'))
+    M1OutpostAI.DisableBase()
+    ScenarioFramework.EndOperationSafety()
+	
+	for k, v in UEFUnits do
+	    if v and not v:IsDead() and (v:GetAIBrain() == ArmyBrains[UEF]) then
+            v:Stop()
+	    end
+	end
+
+    local NISUnits = ScenarioUtils.CreateArmyGroup('UEF', 'PlayerLose')
+    for k, v in NISUnits do
+    	v:SetFireState('HoldFire')
+        IssueAttack({v}, Ship)
+    end
+end
+
+function KillGame()
+    ForkThread(
+        function()
+            ScenarioFramework.EndOperation(ScenarioInfo.OpComplete, ScenarioInfo.OpComplete)
+        end
+    )
+end
+
+function M1NIS()
+	local Engineer1 = ScenarioInfo.UnitNames[Player1]['Engineer_1']
+    local Engineer2 = ScenarioInfo.UnitNames[Player1]['Engineer_2']
+    local Engineer3 = ScenarioInfo.UnitNames[Player1]['Engineer_3']
+    local Engineer4 = ScenarioInfo.UnitNames[Player1]['Engineer_4']
+
+	ScenarioFramework.CreateUnitDamagedTrigger(PlayerLose, ScenarioInfo.Ship, .75)
+
+	WaitSeconds(1)
+
+	Cinematics.EnterNISMode()
+
+	ScenarioFramework.Dialogue(OpStrings.IntroNIS_Dialogue, nil, true)
+	Cinematics.CameraMoveToMarker(ScenarioUtils.GetMarker('Cutscene_01a'), 0)
+	WaitSeconds(.5)
+	Cinematics.CameraTrackEntity(Engineer1, 40, 3)
+
+	WaitSeconds(7)
+
+	Cinematics.CameraMoveToMarker(ScenarioUtils.GetMarker('Cutscene_01b'), 2)
+
+	Cinematics.ExitNISMode()
+
+    if ScenarioInfo.HumanPlayers == 2 then
+        ScenarioFramework.GiveUnitToArmy(ScenarioInfo.Engineer2, Player2)
+        ScenarioFramework.GiveUnitToArmy(ScenarioInfo.Engineer3, Player2)
+    elseif ScenarioInfo.HumanPlayers == 3 then
+        ScenarioFramework.GiveUnitToArmy(ScenarioInfo.Engineer2, Player2)
+        ScenarioFramework.GiveUnitToArmy(ScenarioInfo.Engineer3, Player3)
+    elseif ScenarioInfo.HumanPlayers == 4 then
+        ScenarioFramework.GiveUnitToArmy(ScenarioInfo.Engineer2, Player2)
+        ScenarioFramework.GiveUnitToArmy(ScenarioInfo.Engineer3, Player3)
+        ScenarioFramework.GiveUnitToArmy(ScenarioInfo.Engineer4, Player4)
+    end
+
+	WaitSeconds(1)
+	ForkThread(M1)
+end
+
+
+function M1()
+	ScenarioFramework.CreateTimerTrigger(M1StartUEFScouting, UEFM1Timer[Difficulty])
+
+	local function MissionNameAnnouncement()
+        ScenarioFramework.SimAnnouncement(ScenarioInfo.name, 'mission by [e]JJs_AI')
+    end
+
+    ScenarioFramework.CreateTimerTrigger(MissionNameAnnouncement, 7)
+
+    ScenarioInfo.M1P1 = Objectives.Protect(
+        'primary',                    -- type
+        'incomplete',                   -- complete
+        'Defend Command Ship',  -- title
+        'Defend the Command Ship at all costs. We need it to leave this planet.',  -- description

```

Comment:
It would be better to move the titles and descriptions of the objectives into strings.lua to keep all strings in one file.